packages/experimental/Avatar/src/useAvatar.ts

```diff
@@ -94,3 +97,20 @@ export const validateAlphabeticalCharacters = (name: string): boolean => {
 export const removeTitlesFromName = (words: string[]): string[] => {
   return words.filter((word) => !titles.has(word));
 };
+
+/**
+ * A function which returns hash code for the avatar color
+ *
+ * @param str is idForColor or name
+ * @returns hash code of the color
+ */
+const getHashCode = (str: string): number => {
+  let hashCode = 0;
+  for (let len: number = str.length - 1; len >= 0; len--) {
+    const ch = str.charCodeAt(len);
+    const shift = len % 8;
+    hashCode ^= (ch << shift) + (ch >> (8 - shift)); // eslint-disable-line no-bitwise
+  }
+
+  return hashCode;
+};
```

Comment:
This method I got from Web: https://github.com/microsoft/fluentui/blob/d1b79b83dbffbc0242d8f463715741efec9bf5dd/packages/react-components/react-avatar/src/components/Avatar/useAvatar.tsx#L171-L180

Reply:
I think this may be another area where there will be a question of whether it makes sense to align with Web or Win32. Starting with aligning to web makes sense though, as long as we're consistent with aligning to web with initials logic, too.

Reply:
iOS and macOS both copy the Java hash code algorithm so that we get the same colors across iOS, macOS, and Android

macOS: https://github.com/microsoft/fluentui-apple/blob/66687a715536c95b9443b1e69aac54cb7c96145a/macos/FluentUI/AvatarView/AvatarView.swift#L516
iOS: https://github.com/microsoft/fluentui-apple/blob/66687a715536c95b9443b1e69aac54cb7c96145a/ios/FluentUI/Avatar/Avatar.swift#L379

Another difference I think is that iOS/macOS/Android take some form of primary AND secondary text (i.e: a name and email) and concatenate the strings before hashing them: https://github.com/microsoft/fluentui-apple/blob/66687a715536c95b9443b1e69aac54cb7c96145a/ios/FluentUI/Avatar/Avatar.swift#L326